src/Core/Managed/Shared/TelemetryClient.cs

```diff
@@ -17,14 +17,19 @@
     /// <summary>
     /// Send events, metrics and other telemetry to the Application Insights service.
     /// </summary>
-    public sealed class TelemetryClient
+    public sealed class TelemetryClient : IDisposable
```

Comment:
The problem is with the fact that there is/was no control or cleanup of channels created on the configuration.  Users can define the channel at any time, replace an existing channel, or leave it up to the telemetry client to provide a default.  So when someone creates multiple new telemetry clients, and provides custom config each time, a bunch of non-cleaned up scenarios become possible.